featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -118,6 +118,10 @@ def run(self, instance_ids):
 
         df.index.name = self.entityset[self.feature_set.target_eid].index
         column_list = []
+
+        # Order by instance_ids
+        df = df.reindex(_unique_list(instance_ids))
```

Comment:
to get the unique values in order of appearance, I think `pandas.unique` will be faster 

https://pandas.pydata.org/pandas-docs/version/0.24.2/reference/api/pandas.unique.html